### DISCUSSION TITLE
Beginner level simple question

### DISCUSSION THREAD

**Discussion by chametz17** [dahall/Vanara#396] [Q&A]
I'm trying to import some inherited code from .NET 4.6 to Core.  They were using the DHCPServerAPI package which didn't seem portable to .NET.  So I was trying to use the Vanara one and all the examples I found included a using statement of Vanara.PInvoke.Dhcp.  However, every which way I try to install this I can never get to that path and in turn can't create the DhcpServer object to port the code logic over.  

My gut says I'm missing something very simple.  Any suggestions?  We just started using .NET 7 so I'm not 100% familiar.  I does show in the project file as a package reference.  I tried rolling the project back to 6, but had the same results.  THX.

**Comment by dahall** [comment]
Have you included the NuGet package [Vanara.PInvoke.Dhcp](https://www.nuget.org/packages/Vanara.PInvoke.Dhcp)? That should attach all the libs you need.

**Reply by chametz17** [reply]
Yes ... I use the NuGet manager to install Vanara.PInvoke.Dhcp 3.4.14 to my project.  This shows up in my project file.  You can see my version of .NET and the library.  When I go into the code the only option I have for the USING statement after Vanara.PInvoke is Collections.  

<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net7.0-windows</TargetFramework>
    <Nullable>enable</Nullable>
    <UseWindowsForms>true</UseWindowsForms>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Vanara.PInvoke.Dhcp" Version="3.4.14" />
  </ItemGroup>

</Project>

**Comment by dahall** [comment]
Ah, maybe the following:
```cs
using Vanara.PInvoke;
using static Vanara.PInvoke.Dhcp;
```

**Reply by chametz17** [reply]
I appreciate the response.  I can put both those lines in a cs file no problem.  However, in all the examples I've read the process to start the connection to the server involves a line like this:  

var dhcp = new DhcpServer("192.168.1.1");

This DhcpServer object is not available to me even if I try to full path to it thru Vanara.PInvoke etc.  Are the examples I'm seeing incorrect?  Is there a better way to attach to a server and then be able access leases etc.?

Thanks!

**Comment by dahall** [comment]
Vanara is a set of libraries that include mostly the raw exposure of the native Win32 APIs. There are a few wrappers, like DhcpClient in Vanara.Net, but it is mostly for mimicking the native C++ code like https://github.com/dahall/WinClassicSamplesCS/blob/master/Win7Samples/netds/dhcp/dhcprequest/Program.cs
